src/Interpreters/SessionTracker.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include "ClientInfo.h"
+
+#include <list>
+#include <map>
+#include <memory>
+#include <mutex>
+
+namespace DB
+{
+
+struct SessionInfo
+{
+    explicit SessionInfo(const ClientInfo & client_info, const String & session_id);
```

Comment:
EDIT: I don't get why the session list (l. 21) needs to store full-blown and heavy-weight `ClientInfo` objects. Can't we just store the `session_id` string (perhaps with the `user_id` string to allow throwing a proper exception message) as list node?

Reply:
> EDIT: I don't get why the session list (l. 21) needs to store full-blown and heavy-weight `ClientInfo` objects. Can't we just store the `session_id` string (perhaps with the `user_id` string to allow throwing a proper exception message) as list node?

For the moment it's enough to use only a single integer as a counter of active sessions per user. But later we can add possibility to get current active sessions by SQL like

```sql
SELECT * FROM System.ActiveSessions
```

`System.ActiveSessions` will contain several fields from client info, 
At least it would be good to have: name, `Address`, `ClientInfo::Interface`, `session start time,` `secure/not secure`, `session id`,

Storing the full `ClientInfo `information is redundant, but it's easy in terms of the amount of code involved.

Reply:
I see, that's fine then.

Reply:
I think it would be better to not confuse other developers by adding much unused data from ClientInfo. 
We can add all required fields in further improvements.